pkg/storage/etcd3/watcher.go

```diff
@@ -158,13 +158,17 @@ func (wc *watchChan) sync() error {
 	wc.initialRev = getResp.Header.Revision
 
 	for _, kv := range getResp.Kvs {
-		prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
-		if err != nil {
-			return err
-		}
 		var prevVal []byte
-		if len(prevResp.Kvs) > 0 {
-			prevVal = prevResp.Kvs[0].Value
+		if kv.ModRevision != kv.CreateRevision {
+			prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
+			if err != etcdrpc.ErrCompacted {
+				if err != nil {
+					return err
+				}
+				if len(prevResp.Kvs) > 0 {
+					prevVal = prevResp.Kvs[0].Value
+				}
+			}
```

Comment:
Yeah, I think we need to stop trying to fetch to be compatible with v2. Also, we need to change how parseKV works:

```go
func parseKV(kv *mvccpb.KeyValue, prevVal []byte) *event {
	return &event{
		key:       string(kv.Key),
		value:     kv.Value,
		prevValue: prevVal,
		rev:       kv.ModRevision,
		isDeleted: false,
		isCreated: kv.ModRevision == kv.CreateRevision,
	}
}
```

`isCreated` must always be true.